modules/ollama/local.go

```diff
@@ -0,0 +1,758 @@
+package ollama
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"net"
+	"os"
+	"os/exec"
+	"reflect"
+	"strings"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/network"
+	"github.com/docker/docker/errdefs"
+	"github.com/docker/docker/pkg/stdcopy"
+	"github.com/docker/go-connections/nat"
+
+	"github.com/testcontainers/testcontainers-go"
+	tcexec "github.com/testcontainers/testcontainers-go/exec"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+const (
+	localPort       = "11434"
+	localBinary     = "ollama"
+	localServeArg   = "serve"
+	localLogRegex   = `Listening on (.*:\d+) \(version\s(.*)\)`
+	localNamePrefix = "local-ollama"
+	localHostVar    = "OLLAMA_HOST"
+	localLogVar     = "OLLAMA_LOGFILE"
+)
+
+var (
+	// Ensure localProcess implements the required interfaces.
+	_ testcontainers.Container           = (*localProcess)(nil)
+	_ testcontainers.ContainerCustomizer = (*localProcess)(nil)
+
+	// defaultStopTimeout is the default timeout for stopping the local Ollama process.
+	defaultStopTimeout = time.Second * 5
```

Comment:
bug: we don't need this, as the default value from the terminate options is applied thanks to the NewTerminateOptions call. The default is 10s then

Reply:
Nice catch.